packages/catalog-process/src/services/catalogService.ts

```diff
@@ -665,13 +662,7 @@ export function catalogServiceBuilder(
         throw notValidDescriptor(descriptor.id, descriptor.state);
       }
 
-      const document = (
-        descriptor ? [...descriptor.docs, descriptor.interface] : []
-      ).find((doc) => doc != null && doc.id === documentId);
-
-      if (document === undefined) {
-        throw eServiceDocumentNotFound(eserviceId, descriptorId, documentId);
-      }
```

Comment:
Refactor, to reuse the `retrieveDocument` util function. Minor note: there's no need to check `descriptor ?` before if the descriptor is undefined an error is already thrown above.